gateway-service/src/main/java/org/zowe/apiml/gateway/conformance/ValidateAPIController.java

```diff
@@ -10,96 +10,203 @@
 
     private final VerificationOnboardService verificationOnboardService;
-    private String invalidKey = "org.zowe.apiml.gateway.verifier.wrongServiceId";
+
+    private final DiscoveryClient discoveryClient;
+
 
     /**
-     * Accept serviceID and return the JSON file with appropirate message to show if it is valid
-     *
-     * @param serviceID accepted serviceID to check validation
-     * @return return the JSON file message of whether the serviceID is valid
+     * Accepts serviceID and checks conformance criteria
      *
+     * @param serviceId accepted serviceID to check for conformance
+     * @return 200 if service is conformant, 400 + JSON explanation if not
      */
-    @PostMapping(
-        value = "/validate",
+    @GetMapping(
+        value = "/gateway/conformance/{serviceId}",
         produces = MediaType.APPLICATION_JSON_VALUE
     )
-    public ResponseEntity<Object> checkValidate(@RequestBody String serviceID) {
+    @HystrixCommand
+    public ResponseEntity<String> checkConformance(@PathVariable String serviceId) {
+
+
+        ConformanceProblemsContainer foundNonConformanceIssues = new ConformanceProblemsContainer();
+
+
+        foundNonConformanceIssues.put(problemWithConformance, validateServiceIdFormat(serviceId));
 
-        String message = validator(serviceID);
 
-        if (!message.isEmpty()) {
-            return new ResponseEntity<>(messageService.createMessage(invalidKey, message).mapToApiMessage(), HttpStatus.BAD_REQUEST);
+        if (foundNonConformanceIssues.size() != 0) {
+            return GenerateBadRequestResponseEntity(NonConformantKey, foundNonConformanceIssues);
         }
 
-        return new ResponseEntity<>(HttpStatus.OK);
+        foundNonConformanceIssues.put(problemWithRegistration, checkOnboarding(serviceId));
+
+        if (foundNonConformanceIssues.size() != 0) {     // cant continue if a service isn't registered
+            return GenerateBadRequestResponseEntity(wrongServiceIdKey, foundNonConformanceIssues);
+        }
+
+        List<ServiceInstance> serviceInstances = discoveryClient.getInstances(serviceId);
+        foundNonConformanceIssues.put(problemWithRegistration, instanceCheck(serviceInstances));
+
+        ServiceInstance serviceInstance = serviceInstances.get(0);
+        Map<String, String> metadata = getMetadata(serviceInstance);
 
+        foundNonConformanceIssues.put(problemWithMetadata, metaDataCheck(metadata));
+
+        if (foundNonConformanceIssues.size() != 0) {     // cant continue without metadata
+            return GenerateBadRequestResponseEntity(NoMetadataKey, foundNonConformanceIssues);
+        }
+
+        // Other non conformance checks will be here
+        if (foundNonConformanceIssues.size() != 0) {
+            return GenerateBadRequestResponseEntity(NonConformantKey, foundNonConformanceIssues);
+        }
+
+        return new ResponseEntity<>("{\"message\":\"Service fulfills all checked conformance criteria\"}", HttpStatus.OK);
```

Comment:
It would be nice to include the serviceId in the response message. For better visibility.